src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/PreferHashDataOverComputeHashTests.cs

```diff
@@ -0,0 +1,1487 @@
+using System;
+using System.Security.Cryptography;
+
+public class Test
+{{
+    public static void TestMethod()
+    {{
+        var buffer = new byte[1024];
+        int aboveLine = 20;
+        byte[] digest = {{|#0:{hashType}.Create().ComputeHash({{|#1:buffer|}})|}};
+        int belowLine = 10;
+    }}
+}}
+";
+
+                string csFix = $@"
+using System;
+using System.Security.Cryptography;
+
+public class Test
+{{
+    public static void TestMethod()
+    {{
+        var buffer = new byte[1024];
+        int aboveLine = 20;
+        byte[] digest = {hashType}.HashData(buffer);
+        int belowLine = 10;
+    }}
+}}
+";
+                await TestCSWithoutVariable(
+                    csInput,
+                    csFix,
+                    $"System.Security.Cryptography.{hashType}");
+            }
+        }
+
+        [Fact]
+        public async Task BasicCreateHelperChainCase()
+        {
+            await TestWithType(HashTypeMD5);
+            await TestWithType(HashTypeSHA1);
+            await TestWithType(HashTypeSHA256);
+            await TestWithType(HashTypeSHA384);
+            await TestWithType(HashTypeSHA512);
+
+            static async Task TestWithType(string hashType)
+            {
+                string vbInput = $@"
+Imports System
+Imports System.Security.Cryptography
+
+Public Class Test
+    Public Shared Sub TestMethod()
+        Dim buffer = New Byte(1023) {{}}
+        Dim aboveLine = 20
+        Dim digest As Byte() = {{|#0:{hashType}.Create().ComputeHash({{|#1:buffer|}})|}}
+        Dim belowLine = 10
+    End Sub
+End Class
+";
+
+                string vbFix = $@"
+Imports System
+Imports System.Security.Cryptography
+
+Public Class Test
+    Public Shared Sub TestMethod()
+        Dim buffer = New Byte(1023) {{}}
+        Dim aboveLine = 20
+        Dim digest As Byte() = {hashType}.HashData(buffer)
+        Dim belowLine = 10
+    End Sub
+End Class
+";
+                await TestVBWithoutVariable(
+                    vbInput,
+                    vbFix,
+                    $"System.Security.Cryptography.{hashType}");
+            }
+        }
+
+        [Fact]
+        public async Task CSharpCreateHelperNoUsingStatementBailOutNoFixCase()
+        {
+            await TestWithType(HashTypeMD5);
+            await TestWithType(HashTypeSHA1);
+            await TestWithType(HashTypeSHA256);
+            await TestWithType(HashTypeSHA384);
+            await TestWithType(HashTypeSHA512);
+
+            static async Task TestWithType(string hashType)
+            {
+                string csInput = $@"
+using System;
+using System.Security.Cryptography;
+
+public class Test
+{{
+    public static void TestMethod()
+    {{
+        var buffer = new byte[1024];
+        var hasher = {hashType}.Create();
+        int aboveLine = 20;
+        byte[] digest = hasher.ComputeHash(buffer);
+        int belowLine = 10;
+        byte[] digest2 = hasher.ComputeHash(buffer);
```

Comment:
Hmm, I think it's okay actually. 
For the fixer, I have an idea where if there's more than 1 ComputeHash, it will just replace ComputeHash.
If it's the last/only one, the fixer can delete the `Create` method.

However, would there be an issue with the `Fix All` function?

Reply:
It seems I have to add a Fix All

Reply:
I've opt to disable FixAll

Reply:
> However, would there be an issue with the Fix All function?
> I've opt to disable FixAll

Not sure, though i am OK with disabling, ping @jmarolf @sharwell

Reply:
I gave doing a custom fixallprovider a go but I'm running into differences in the formatting done. How do i replicate the formatting done in a normal codefix?
Is there an api to pass the changed document? Or is creating the SourceText the way in a fixallprovider?

Reply:
> I gave doing a custom fixallprovider a go but I'm running into differences in the formatting done. How do i replicate the formatting done in a normal codefix?
Is there an api to pass the changed document? Or is creating the SourceText the way to go in a fixallprovider?

I am not familiar with custom fixallprovider, pinging @mavasani @jmarolf @sharwell 

In case there is no such API and it has only issue in formatting i think that is OK

Reply:
There's a case that is complicated for the fixall, I'm still looking into that.